File: FreelanceMarketplace/FreelanceMarketplace/GraphQL/Types/UserType.cs
using FreelanceMarketplace.Models;
using GraphQL.Types;

namespace FreelanceMarketplace.GraphQL.Types
{
    public class UserType : ObjectGraphType<Users>
    {
        public UserType()
        {
            Field(x => x.Id);
            Field(x => x.Username);
            Field(x => x.PasswordHash);
            Field(x => x.Email);
            Field(x => x.Role);
            Field(x => x.IsEmailConfirmed);
            Field(x => x.Status);
            Field(x => x.CreateAt);
            Field<ListGraphType<ProjectType>>("projects");
            Field<ListGraphType<RefreshTokenType>>("refreshTokens");
            Field<ListGraphType<ProjectType>>("uploadedImages");
            Field<UserProfileType>("userProfile");
            Field<ReviewType>("review");
        }
    }
}

File: FE/src/types/UserType.ts
import { ProjectType } from "./ProjectType";
import { ReviewType } from "./ReviewType";

export type User = {
    id: number;
    username: string;
    passwordHash: string;
    email: string;
    role: string;
    isEmailConfirmed: boolean;
    createAt: string;
    status: string;
    userProfile: UserProfileType;
    projects?: ProjectType[];
    earning?: number;
    reviews: ReviewType[];
}

export interface UserProfileType {
    id: number;
    rating: number;
    company?: string;
    location?: string;
    phone?: string;
    birthday?: Date;
    gender?: string;
    bio?: string;
    skill?: string;
    avatar?: string;
    Industry?: string;
    status: string;
};

export interface UserProfileInput {
    rating: number;
    company: string;
    location: string;
    phone: string;
    birthday: Date;
    gender: string;
    bio: string;
    skill: string;
    avatar: string;
    industry: string;
}